src/formatDate.js

```diff
@@ -8,7 +8,50 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const dateParts = date.split(fromFormat[3]);
+  let year;
+  let month;
+  let day;
+  const formattedDate = [];
+
+  for (let i = 0; i < fromFormat.length; i++) {
+    if (fromFormat[i] === 'YYYY') {
+      year = dateParts[i];
+    }
+
+    if (fromFormat[i] === 'YY') {
+      year = dateParts[i] < 30 ? `20${dateParts[i]}` : `19${dateParts[i]}`;
+    }
+
+    if (fromFormat[i] === 'MM') {
+      month = dateParts[i];
+    }
+
+    if (fromFormat[i] === 'DD') {
+      day = dateParts[i];
+    }
+  }
+
+  for (const format of toFormat) {
+    switch (format) {
+      case 'YYYY':
+        formattedDate.push(year);
+        break;
+
+      case 'YY':
+        formattedDate.push(year.slice(-2));
```

Comment:
There is an implicit assumption that `year` will always be a string of at least two characters. While this should be the case for valid dates, adding a check to ensure that `year` has the expected length before slicing could prevent potential errors.